### ISSUE TITLE
CSV Data base fine

### ISSUE THREAD

**Issue by Lilaz31** [ipb-halle/MetFrag#23]
[Test DB csv-Aurones.csv](https://github.com/ipb-halle/MetFrag/files/11955329/Test.DB.csv-Aurones.csv)
Hi,
I wated to test some molecules for our MS data but the CSV that I created is not accepted. 
Can you please check the database file and tell where are errors? 
Thank you. 
Lila

**Comment by schymane** [comment]
I thought at first this was a ";" vs "," issue, but rather it looks like your InChIs are not protected (with a quote), when I open the file the InChIs are spread over multiple columns, which will affect the alignment. 
![image](https://github.com/ipb-halle/MetFrag/assets/4070141/a8d0342a-f592-484e-8190-bd57b6e2059e)

In Notepad++ all separators are ",", but the InChIs appear without "", which means they will be split, as seen above.
![image](https://github.com/ipb-halle/MetFrag/assets/4070141/e100540f-591f-4851-a34d-2995c697d56c)

Note also that you should not need to produce the InChIKey1, 2 and 3 columns, just the InChIKey column alone will suffice (MetFrag does the splitting internally for it's own purposes). I hope this helps! If in doubt, quote-protect everything (depending on how you create the file?). The same issue will also affect some "Name" entries, commas are also common in some chemical names.

**Comment by Lilaz31** [comment]
thank you for your help. 

[Test DB csv-Aurones_bis.csv](https://github.com/ipb-halle/MetFrag/files/11956562/Test.DB.csv-Aurones_bis.csv)
I corrected my database but it still not working!

**Comment by schymane** [comment]
yes now you have ";" separators and not ",", which seems to also cause problems (now it is only separating at the protected commas). 
![image](https://github.com/ipb-halle/MetFrag/assets/4070141/e7413db2-6b47-4785-a606-0b5fa95defc1)

However, just substituting ";" with "," also didn't work for me, 
[Test.DB.csv-Aurones_bis_ES.csv](https://github.com/ipb-halle/MetFrag/files/11956706/Test.DB.csv-Aurones_bis_ES.csv)
![image](https://github.com/ipb-halle/MetFrag/assets/4070141/50340f82-b14e-4ebf-968f-f1cbd4d49b10)

but this file finally did: 
[test_CSV.csv](https://github.com/ipb-halle/MetFrag/files/11956690/test_CSV.csv)

![image](https://github.com/ipb-halle/MetFrag/assets/4070141/fb52e3ff-92c7-47d2-8b3d-f1cc3f7e830b)

...and also just "Name" as you had it, not "CompoundName" also works
[test_CSV2.csv](https://github.com/ipb-halle/MetFrag/files/11956710/test_CSV2.csv)

![image](https://github.com/ipb-halle/MetFrag/assets/4070141/be038518-ee92-4723-bd2c-f67a3fead44f)

**Comment by Lilaz31** [comment]
Thanks a lot. I am a chemist and not a computer scientist and it is a bit complicated to grasp everything.

























































Thank you very much.

**Comment by Lilaz31** [comment]
Is there another software for creating databases? I wanted to add some molecules to my previous database following the same syntax but it's not working. So maybe it's a compatibility problem with the excel software? When I open the file in notepad, I didn't detected any mistake. So how can i generate these kind of files without mistakes? 
Here the file with two more molecules (A19 and A20):
[test_CSV-20molecules.csv](https://github.com/ipb-halle/MetFrag/files/11957796/test_CSV-20molecules.csv)

**Comment by Lilaz31** [comment]
I think I really have a compatibility issue when converting to CSV from Excel. I just redo the same operation but starting from the notepad and there it works well.

**Comment by schymane** [comment]
Yes it could be something to do with your locale settings in excel ... I frequently create test files in excel and don't often encounter issues, although on occasions I've had to copy everything into a new file, save as CSV under a new name, and only the new one works. I run in an English locale with Australian default settings for decimals, commas etc. Some language settings cause issues with CSV (not sure if this applies in your case)
If you really run into issues then the "PSV" format is always an option, there you have to save with a "|" (pipe, hence P) as the separator, but this might not be easy to do in excel (it is easier if you create the file programmatically). 
Usually excel should automatically protect entries with commas like the InChI though, it is strange that it does not do that for you. 
Sometimes it is trial and error - I have been creating database files for years and still encounter strange issues! So you are not alone. One other way to try (if you have issues next time) is to download an existing candidate file as CSV and add your entries to the bottom, and reupload it to see if that works (you can delete the three extra InChIKey columns - but you will need the full InChIKey column still present).
